ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/strategy/transferstrategy/AbstractTransferStrategy.java

```diff
@@ -117,20 +127,21 @@ public void transfer() throws IOException {
     }
 
     private void transferAllJobBlobs() throws IOException {
+        final Predicate transferPredicate = () -> ! canceled;
         final AtomicInteger numBlobsRemaining = new AtomicInteger(jobState.numBlobsInJob());
 
-        while (!Thread.currentThread().isInterrupted() && numBlobsRemaining.get() > 0) {
+        while ( ! Thread.currentThread().isInterrupted() && transferPredicate.test() && numBlobsRemaining.get() > 0) {
             try {
-                final Iterable<JobPart> jobParts = jobPartsNotAlreadyTransferred();
+                final Iterable<JobPart> jobParts = jobPartsNotAlreadyTransferred(transferPredicate);
 
                 final int numJobParts = Iterables.size(jobParts);
 
                 if (numJobParts <= 0) {
                     break;
                 }
 
-                final CountDownLatch countDownLatch = new CountDownLatch(numJobParts);
-                transferJobParts(jobParts, countDownLatch, numBlobsRemaining);
+                final CountDownLatch countDownLatch = new CountDownLatch(1);
```

Comment:
We should probably rename this countDownLatch to something like, `jobCompletionWait` I'm a little confused though, why was the count here changed?  I see a couple places where we decrement it in the `transferJobParts` code, but I've not yet convinced myself that it only ever decrements it once.